theseus/embodied/kinematics/kinematics_model.py

```diff
@@ -90,25 +108,20 @@ def forward_kinematics(
             )
 
         # Compute forward kinematics for all links
-        fk_output = self.drm_model.compute_forward_kinematics_all_links(
-            joint_states_input
-        )
-
-        link_poses: Dict[str, LieGroup] = {}
-        for link_name in self.drm_model.get_link_names():
-            pos, quat = fk_output[link_name]
-            quat_processed = self._postprocess_quaternion(quat)
+        fk_output = self.fk(joint_states_input)
 
-            link_poses[link_name] = SE3(
-                x_y_z_quaternion=torch.cat([pos, quat_processed], dim=-1)
-            )
+        link_poses = {}
+        for i, name in enumerate(self.link_names):
+            link_poses[name] = SE3(tensor=fk_output[i], strict_checks=False)
 
         # Compute jacobians
         if jacobians is not None:
-            for link_name in jacobians.keys():
-                jac_lin, jac_rot = self.drm_model.compute_endeffector_jacobian(
-                    joint_states_input, link_name
-                )
-                jacobians[link_name] = torch.cat([jac_lin, jac_rot], dim=-2)
+            jacs_out = (
+                self.jfk_b(joint_states_input)
```

Comment:
If `jacobians` is not `None`, we should compute `jacobians` and `link_poses` together from `jfk_b` or `jfk_s`.